samples/python/classification_sample_async/classification_sample_async.py

```diff
@@ -2,164 +2,128 @@
 
-            log.info(f'Infer request {i} returned {infer_status}')
+# --------------------------- Step 4. Apply preprocessing -------------------------------------------------------------
+    ppp = PrePostProcessor(model)
 
-            if infer_status != StatusCode.OK:
-                return -2
+    # 1) Set input tensor information:
+    # - input() provides information about a single model input
+    # - precision of tensor is supposed to be 'u8'
+    # - layout of data is 'NHWC'
+    ppp.input().tensor() \
+        .set_element_type(Type.u8) \
+        .set_layout(Layout('NHWC'))  # noqa: N400
 
-            # Read infer request results from buffer
-            res = exec_net.requests[i].output_blobs[out_blob].buffer
-            # Change a shape of a numpy.ndarray with results to get another one with one dimension
-            probs = res.reshape(num_of_classes)
-            # Get an array of args.number_top class IDs in descending order of probability
-            top_n_idexes = np.argsort(probs)[-args.number_top :][::-1]
+    # 2) Here we suppose model has 'NCHW' layout for input
+    ppp.input().model().set_layout(Layout('NCHW'))
 
-            header = 'classid probability'
-            header = header + ' label' if args.labels else header
+    # 3) Set output tensor information:
+    # - precision of tensor is supposed to be 'f32'
+    ppp.output().tensor().set_element_type(Type.f32)
 
-            log.info(f'Image path: {args.input[i]}')
-            log.info(f'Top {args.number_top} results: ')
-            log.info(header)
-            log.info('-' * len(header))
+    # 4) Apply preprocessing modifing the original 'model'
+    model = ppp.build()
 
-            for class_id in top_n_idexes:
-                probability_indent = ' ' * (len('classid') - len(str(class_id)) + 1)
-                label_indent = ' ' * (len('probability') - 8) if args.labels else ''
-                label = labels[class_id] if args.labels else ''
-                log.info(f'{class_id}{probability_indent}{probs[class_id]:.7f}{label_indent}{label}')
-            log.info('')
+# --------------------------- Step 5. Loading model to the device -----------------------------------------------------
+    log.info('Loading the model to the plugin')
+    compiled_model = core.compile_model(model, args.device)
 
-            output_queue.remove(i)
+# --------------------------- Step 6. Create infer request queue ------------------------------------------------------
+    log.info('Starting inference in asynchronous mode')
+    infer_queue = AsyncInferQueue(compiled_model, len(input_tensors))
+    infer_queue.set_callback(completion_callback)
 
-        if len(output_queue) == 0:
-            break
+# --------------------------- Step 7. Do inference --------------------------------------------------------------------
+    for i, input_tensor in enumerate(input_tensors):
+        infer_queue.start_async({0: input_tensor}, args.input[i])
```

Comment:
In fact you are able to make it even more python-friendly as `start_async(...)` is called from here:
https://github.com/openvinotoolkit/openvino/blob/eb94fe27a61f211aec453f6f0721f4f61703e263/src/bindings/python/src/openvino/runtime/ie_api.py#L84
And underlaying function is converting numpy arrays to Tensors in runtime:
https://github.com/openvinotoolkit/openvino/blob/eb94fe27a61f211aec453f6f0721f4f61703e263/src/bindings/python/src/openvino/runtime/ie_api.py#L35
So you can change this loop to something like:
```suggestion
    for i, image in enumerate(resized_images):
        infer_queue.start_async({0: np.expand_dims(image, 0)}, args.input[i])
```
However the change could have effect on performance of this loop, as all numpy actions and Tensor creations will be done here. What do you win? Removal of lines 88-89 where Tensors are being created, or even more lines if you introduce resizing in this loop as well. Keep in mind - this is only example to show how API can work - for performance we have benchmark_app.

Reply:
done